docs/elements/sageone/authenticate.md

```diff
@@ -0,0 +1,275 @@
+| apiSecret |  The client secret obtained from registering your app with the provider.  This is the **Client Secret** that you noted at the end of the [Service Provider Setup section](setup.html).   |
+| callbackUrl | The URL that will receive the code from the vendor to be used to create an element instance. This is the **Callback URL** that you noted at the end of the [Service Provider Setup section](setup.html).  |
+
+#### Example cURL
+
+```bash
+curl -X GET
+-H 'Content-Type: application/json'
+'https://api.cloud-elements.com/elements/api-v2/elements/{{page.elementKey}}/oauth/url?apiKey=fake_api_key&apiSecret=fake_api_secret&callbackUrl=https://www.mycoolapp.com/auth&state={{page.elementKey}}'
+```
+
+#### Example Response
+
+Use the `oauthUrl` in the response to allow users to authenticate with the vendor.
+
+```json
+{
+"element": "{{page.elementKey}}",
+"oauthUrl": "https://www.sageone.com/oauth2/auth?scope=full_access&response_type=code&redirect_uri=https%3A%2F%2Fhttpbin.org%2Fget&state=sageone&client_id=b8cb21f24d3f1a2d8c5d"
+}
+```
+
+### Authenticating Users and Receiving the Authorization Grant Code
+
+Provide the response from the previous step to the users. After they authenticate, {{page.heading}} provides the following information in the response:
+
+* code
+* country
+* state
+
+| Response Parameter | Description   |
+| :------------- | :------------- |
+| code | The Authorization Grant Code required by Cloud Elements to retrieve the OAuth access and refresh tokens from the endpoint.|
+| country | The country code associated with your account at {{page.heading}} |
+| state | A customizable identifier, typically the element key (`{{page.elementKey}}`) . |
+
+{% include note.html content="If the user denies authentication and/or authorization, there will be a query string parameter called <code>error</code> instead of the <code>code</code> parameter. In this case, your application can handle the error gracefully." %}
+
+### Authenticating the Element Instance
+
+Use the `/instances` endpoint to authenticate with {{page.heading}} and create an element instance. If you are configuring events, see the [Events section](events.html).
+
+{% include note.html content="The endpoint returns an Element token upon successful completion. Retain the token for all subsequent requests involving this element instance.  " %}
+
+To create an element instance:
+
+1. Construct a JSON body as shown below (see [Parameters](#parameters)):
+
+    ```json
+            {
+              "element": {
+                "key": "{{page.elementKey}}"
+              },
+              "providerData": {
+                "code": "<AUTHORIZATION_GRANT_CODE>"
+              },
+              "configuration": {
+                "oauth.api.key": "<CLIENT_ID>",
+                "oauth.api.secret": "<CLIENT_SECRET>",
+                "signature.secret": "<SIGNING_SECRET>",
+              	"apim.subscription.key": "<SUBSCRIPTION_PRIMARY_KEY_or_APIM_SUBSCRIPTION_KEY>",
```

Comment:
Whats the apim subscription key, is this the grant code from the previous step, or is that different key that we need to pass in?